Adds Status to BlogPosts

demirmusa opened the pull request:
It currently implement Draft and Puslished status. Closes #11963
(https://github.com/abpframework/abp/issues/11964 will inherit that pr and implement new status.)

## Note!!
- [x] Waiting for https://github.com/abpframework/abp/pull/12048 since it adds filter to appservice and that pr extends it. That's why first merge https://github.com/abpframework/abp/pull/12048 then that pr.

demirmusa committed:
add status to BlogPost

demirmusa committed:
add migration

demirmusa committed:
add publish

demirmusa committed:
add create and publish

demirmusa committed:
add draft

demirmusa commented:
- [x] Waiting for unit tests

demirmusa committed:
Add BlogPostStatus to BlogPostDto

demirmusa committed:
Use status filter in GetCountAsync method in public blog post GetAllListAsync

demirmusa committed:
fix tests

demirmusa committed:
add tests

rqx110 commented on modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Blogs/BlogPostAdminAppService.cs:
```diff
@@ -110,4 +113,30 @@ public virtual async Task DeleteAsync(Guid id)
+        await CurrentUnitOfWork.SaveChangesAsync();
+        
+        await PublishAsync(blogPost.Id);
+        blogPost.Status = BlogPostStatus.Published;
```
Duplicated! 
`blogPost.Status = BlogPostStatus.Published;` it has already been seted in `await PublishAsync(blogPost.Id);`

demirmusa replied:
No is it not. It is just a dto which values filled before we change entity's status value in `PublishAsync`. It contains previous value.

rqx110 replied:
ok,I get it

demirmusa replied:
Thank you for taking the time to contribute @rqx110

enisn commented on modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Blogs/BlogPostAdminAppService.cs:
```diff
@@ -110,4 +113,30 @@ public virtual async Task DeleteAsync(Guid id)
+    public virtual async Task DraftAsync(Guid id)
+    {
+        var blogPost = await BlogPostRepository.GetAsync(id);
+        blogPost.SetDraft();
+    }
```
Calling update method of repository here might be better since MongoDB doesn't have entity tracking system and it won't be updated in MongoDB at the end of uow. In other hand, transaction can be disabled for MongoDB, in that case, this entity won't be updated
 
```suggestion
        blogPost.SetDraft();
        await BlogPostRepository.UpdateAsync(blogPost);
    }
```

rqx110 replied:
seems also need same changes in PublishAsync method

enisn replied:
Yeah, same for **PublishAsync()** method too

demirmusa replied:
Thank you. It is resolved in [03e7905](https://github.com/abpframework/abp/pull/12073/commits/03e790547fbc0fb02589edb216218b657d9367d5)

demirmusa committed:
update blog post when it's status changed

It is required for MongoDb.
It is requested in https://github.com/abpframework/abp/pull/12073#discussion_r835100870

demirmusa committed:
use kebab-case in route

demirmusa committed:
change route

demirmusa committed:
add forgetten localization key

demirmusa committed:
list author only has published posts

enisn committed:
Merge branch 'dev' into pr/11963

enisn committed:
Fix wrong conflict merging from dev

enisn committed:
Merge branch 'dev' into pr/11963
